io_buffer.c

```diff
@@ -2695,6 +2737,21 @@ rb_io_buffer_pwrite(VALUE self, VALUE io, rb_off_t from, size_t length, size_t o
     return rb_thread_io_blocking_region(io_buffer_pwrite_internal, &argument, descriptor);
 }
 
+/*
+ *  call-seq: pwrite(io, from, length, [offset]) -> numbytes
+ *
+ *  Writes +length+ bytes from buffer into +io+, starting at
+ *  +offset+ in the buffer.
+ *
+ *  If +offset+ is not given, the bytes are taken from the beginning
```

Comment:
I would suggest adding that the output file will be zero filled if the offset is beyond the end of the current file size.